### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by vanyachyzh**
[DEMO LINK](https://vanyachyzh.github.io/react_movies-list-add-form/)

### CODE REVIEW

File: src/components/NewMovie/NewMovie.tsx [hunk 1]
```diff
@@ -1,45 +1,83 @@
...
-    <form className="NewMovie" key={count}>
+    <form
+      className="NewMovie"
+      key={count}
+      onSubmit={(e) => {
+        e.preventDefault();
+        setCount(count + 1);
+        onAdd(movie);
+        setMovie(emptyMovie);
+      }}
```
Comment (VitaliyBondarenko1982) [inline src/components/NewMovie/NewMovie.tsx:40]: make function for it and pass here reference.
check please other cases in project

File: src/components/NewMovie/NewMovie.tsx [hunk 2]
```diff
@@ -1,45 +1,83 @@
...
+  title: '',
+  description: '',
+  imdbUrl: '',
+  imgUrl: '',
+  imdbId: '',
+};
+
+export const NewMovie: React.FC<Props> = ({ onAdd }) => {
+  const [count, setCount] = useState(0);
+  const [movie, setMovie] = useState(emptyMovie);
```
Comment (VitaliyBondarenko1982) [inline src/components/NewMovie/NewMovie.tsx:19]:
```suggestion
  const [formInputs, setFormInputs] = useState(initialFormInputs);
```

Comment (VitaliyBondarenko1982) [reply src/components/NewMovie/NewMovie.tsx:19]: suggesting more meaningful naming

File: src/components/NewMovie/NewMovie.tsx [hunk 3]
```diff
@@ -1,45 +1,83 @@
...
+  // eslint-disable-next-line object-curly-newline
+  const { title, description, imdbId, imdbUrl, imgUrl } = movie;
+  const isDisabled: boolean = !title || !imdbId || !imdbUrl || !imgUrl;
+
+  const setProperty = (property: string, value: string) => {
+    setMovie(prevState => ({
+      ...prevState,
+      [property]: value,
+    }));
+  };
```
Comment (VitaliyBondarenko1982) [inline src/components/NewMovie/NewMovie.tsx:29]:
```suggestion
  const onChange = (event: React.ChangeEvent<HTMLInputElement>) => {
   const { name, value } = event.target;
   
    setMovie(prevState => ({
      ...prevState,
      [name]: value,
    }));
  };
```

use name attribute to change inputField. and pass in TextField only reference to onChange function

File: src/components/NewMovie/NewMovie.tsx [hunk 4]
```diff
@@ -1,45 +1,83 @@
...
+    >
       <h2 className="title">Add a movie</h2>
 
       <TextField
         name="title"
         label="Title"
-        value=""
-        onChange={() => {}}
+        value={title}
+        onChange={value => setProperty('title', value)}
```
Comment (VitaliyBondarenko1982) [inline src/components/NewMovie/NewMovie.tsx:48]:
```suggestion
        onChange={onChange}
```

Comment (VitaliyBondarenko1982) [reply src/components/NewMovie/NewMovie.tsx:48]: Same for other TextField

**Review by VitaliyBondarenko1982 — CHANGES_REQUESTED** [review]

File: src/components/NewMovie/NewMovie.tsx [hunk 5]
```diff
@@ -1,45 +1,87 @@
...
+  const onChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const { name, value } = event.target;
+
+    setFormInputs(prevState => ({
+      ...prevState,
+      [name]: value,
+    }));
+  };
+
+  const sendForm = (e: React.SyntheticEvent) => {
```
Comment (etojeDenys) [inline src/components/NewMovie/NewMovie.tsx:33]: `React.FormEvent<HTMLFormElement>`

**Review by etojeDenys — CHANGES_REQUESTED** [review]
almost done

**Pull request closed without merging**